spec/integration/s3_spec.rb

```diff
@@ -0,0 +1,100 @@
+require "logstash/devutils/rspec/spec_helper"
+require "logstash/outputs/s3"
+require 'socket'
+require "aws-sdk"
+require "fileutils"
+require "stud/temporary"
+
+describe LogStash::Outputs::S3, :integration => true, :s3 => true do
+  let!(:minimal_settings)  {  { "access_key_id" => ENV['AWS_ACCESS_KEY_ID'],
+                               "secret_access_key" => ENV['AWS_SECRET_ACCESS_KEY'],
+                               "bucket" => ENV['AWS_LOGSTASH_TEST_BUCKET'], 
+                               "temporary_directory" => Stud::Temporary.pathname('temporary_directory') }}
+
+  let!(:s3_object) do
+      s3output = LogStash::Outputs::S3.new(minimal_settings)
+      s3output.register
+      s3output.s3
+  end
+
+  after(:all) do
+    delete_matching_keys_on_bucket('studtmp')
+    delete_matching_keys_on_bucket('my-prefix')
+  end
+
+  def delete_matching_keys_on_bucket(prefix)
```

Comment:
you can "move" this functions into a support module, so they not mix with the test cases.